mri/reconstructors/self_calibrating.py

```diff
@@ -0,0 +1,192 @@
+                n_coils=1,
+                verbose=bool(verbose >= 30),
+            )
+        # Ensure that we are in right multichannel config
+        if linear_op.n_coils != 1:
+            raise ValueError("The value of n_coils for linear operation must "
+                             "be 1 for Self-Calibrating reconstruction!")
+        if gradient_formulation == 'analysis':
+            grad_class = GradSelfCalibrationAnalysis
+        elif gradient_formulation == 'synthesis':
+            grad_class = GradSelfCalibrationSynthesis
+        super(SelfCalibrationReconstructor, self).__init__(
+            fourier_op=fourier_op,
+            linear_op=linear_op,
+            regularizer_op=regularizer_op,
+            gradient_formulation=gradient_formulation,
+            grad_class=grad_class,
+            lipschitz_cst=lipschitz_cst,
+            lips_calc_max_iter=lips_calc_max_iter,
+            num_check_lips=num_check_lips,
+            init_gradient_op=False,
+            verbose=verbose,
+        )
+        self.smaps_gridding_method = smaps_gridding_method
+        self.smaps_extraction_mode = smaps_extraction_mode
+        if type(kspace_portion) == float:
+            self.kspace_portion = (kspace_portion,) * \
+                                  self.fourier_op.samples.shape[-1]
+        else:
+            self.kspace_portion = kspace_portion
+        if len(self.kspace_portion) != self.fourier_op.samples.shape[-1]:
+            raise ValueError("The k-space portion size used to estimate the" +
+                             "sensitivity information is not aligned with" +
+                             " the input dimension")
+        self.n_jobs = n_jobs
+
+    def reconstruct(self, kspace_data, optimization_alg='pogm', x_init=None,
+                    num_iterations=100, recompute_smaps=True, **kwargs):
+        """ This method calculates operator transform.
+        Parameters
+        ----------
+        kspace_data: np.ndarray
+            the acquired value in the Fourier domain.
+            this is y in above equation.
+        optimization_alg: str (optional, default 'pogm')
+            Type of optimization algorithm to use, 'pogm' | 'fista' |
+            'condatvu'
+        x_init: np.ndarray (optional, default None)
+            input initial guess image for reconstruction
+        num_iterations: int (optional, default 100)
+            number of iterations of algorithm
+        recompute_smaps: bool (optional, default False)
+            A boolean value to check if the Smaps must be recalculated.
+            Note that this would recompute the lipchitz constant.
+            This must be set to True if you want the Smaps to be updated
+            in this reconstruction. The first reconstruction would need
+            this to be True.
+        """
+        if self.fourier_op.n_coils != kspace_data.shape[0]:
+            raise ValueError("The provided number of coil (n_coils) do not "
```

Comment:
Could you test that this error actually raises ?

Reply:
Adding and also moving this check to base class as we want this on every incoming data